### PR TITLE
[train] New persistence mode: Sanity-check release test

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

This PR creates a multinode version of the existing `test_new_persistence` CI test. It tests the new persistence mode with checkpointing to NFS and s3/gcs.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
(for @can-anyscale to review)

**Review by matthewdeng — APPROVED** [review]

File: release/ray_release/byod/byod_train_persistence_test.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+# This script is used to build an extra layer on top of the base anyscale/ray image
+# to run the train_multinode_persistence test.
+
+set -exo pipefail
+
+pip3 install -U torch fsspec s3fs gcsfs pyarrow>=9.0.0 pytest
```
Comment (can-anyscale) [inline release/ray_release/byod/byod_train_persistence_test.sh:7]: I would recommend to use the managed library dependencies in https://www.notion.so/anyscale-hq/OSS-Release-Tests-9e0b126f7dd542aa80e4a6f936b8267f to install pip

post_command is used for other test setup that does not install libraries

Comment (can-anyscale) [reply release/ray_release/byod/byod_train_persistence_test.sh:7]: Or maybe add a TODO to move this to managed dependencies, according to https://www.notion.so/anyscale-hq/OSS-Release-Tests-9e0b126f7dd542aa80e4a6f936b8267f, and deal with it soon in the future

**Review by can-anyscale — APPROVED** [review]

File: release/ray_release/byod/byod_train_persistence_test.sh (hunk 1 above)
Comment (matthewdeng) [reply release/ray_release/byod/byod_train_persistence_test.sh:7]: Sounds good we will follow up on this!

**Pull request merged**